packages/vaadin-virtual-list/src/virtualizer-iron-list-adapter.js

```diff
@@ -103,11 +103,13 @@ export class IronListAdapter {
   }
 
   flush() {
-    this._resizeHandler();
-    flush();
-    this._scrollHandler();
-    this.__scrollReorderDebouncer && this.__scrollReorderDebouncer.flush();
-    this.__debouncerWheelAnimationFrame && this.__debouncerWheelAnimationFrame.flush();
+    if (this.scrollTarget.offsetHeight) {
+      this._resizeHandler();
+      flush();
+      this._scrollHandler();
+      this.__scrollReorderDebouncer && this.__scrollReorderDebouncer.flush();
+      this.__debouncerWheelAnimationFrame && this.__debouncerWheelAnimationFrame.flush();
+    }
```

Comment:
**nit:** Seems that the logic for `offsetHeight === 0` will likely not appear over time so that I would suggest reducing the nesting level.

```suggestion
    // The scroll target is hidden.
    if (this.scrollTarget.offsetHeight === 0) {
      return;
    }
    
    this._resizeHandler();
    flush();
    this._scrollHandler();
    this.__scrollReorderDebouncer && this.__scrollReorderDebouncer.flush();
    this.__debouncerWheelAnimationFrame && this.__debouncerWheelAnimationFrame.flush();
```